plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableParameter.java

```diff
@@ -184,10 +190,11 @@ private Object uncoerce(Object o, Type type) {
         } else if (o != null && !o.getClass().getName().startsWith("java.")) {
             try {
                 // Check to see if this can be treated as a data-bound struct.
-                Map<String, Object> nested = DescribableModel.uninstantiate_(o);
+                UninstantiatedDescribable nested = DescribableModel.uninstantiate2_(o);
                 if (type != o.getClass()) {
-                    nested.put(CLAZZ, o.getClass().getSimpleName());
+                    nested.setKlass(o.getClass().getSimpleName());
                 }
+                nested.setSymbol(symbolOf(o));
```

Comment:
So this is something I forgot to mention yesterday. Do we now require the symbol even on elements of homomorphic collections? So

```groovy
checkout(git(userRemoteConfigs: [
  userRemoteConfig(url: '…', credentialsId: '…')]))
```

By contrast, `$class` can be omitted in such cases; the fact that the list element is a `Map` suffices to infer its type as `UserRemoteConfig`:

```groovy
checkout([$class: 'GitSCM', userRemoteConfigs: [
  [url: '…', credentialsId: '…']]])
```

If we wanted to carry over that simplification, we could either continue to access `Map`s in mixed usage:

```groovy
checkout(git(userRemoteConfigs: [
  [url: '…', credentialsId: '…']]))
```

or introduce a dummy function which just creates an `UninstantiatedDescribable` without a symbol set:

```groovy
checkout(git(userRemoteConfigs: [
  _(url: '…', credentialsId: '…')]))
```